### ISSUE TITLE
Deadlock when setting Metadata in cache

### ISSUE THREAD

**Issue by Rotivity** [object/Simple.OData.Client#431]
Hi,

We need to use GetAwaiter().GetResult() in one of our projects on a high level due to architectural constraints. However, after the Metadata request is executed the application hangs for a while, sometimes indefinitely. I traced it back to a missing 'ConfigureAwait(false)' in Session.cs line 91. After using the source code directly in our project and adding the ConfigureAwait(false) all is well. I can't think of a reason this would be intentional so I assume this is just a mistake. Will this be fixed in the next release?

File: Simple.OData.Client\Simple.OData.Client.Core\Session.cs
Function: Task<IODataAdapter> ResolveAdapterAsync(CancellationToken cancellationToken)
Line: 91
```C#
87                            return cache;
88                        });
89            }
90
91            await this.MetadataCache.Resolved; // missing ConfigureAwait
92
93            if (this.Settings.PayloadFormat == ODataPayloadFormat.Unspecified)
94                this.Settings.PayloadFormat = this.Adapter.DefaultPayloadFormat;
95
96            return this.Adapter;
97        }
```

```C#
87                            return cache;
88                        });
89            }
90
91            await this.MetadataCache.Resolved.ConfigureAwait(false); // fix
92
93            if (this.Settings.PayloadFormat == ODataPayloadFormat.Unspecified)
94                this.Settings.PayloadFormat = this.Adapter.DefaultPayloadFormat;
95
96            return this.Adapter;
97        }
```

Regards,
Robin

**Comment by bmeredith** [comment]
This is an issue we're running into as well after updating to 4.28.0.

**Comment by tomashevich** [comment]
Probably we has this problem too. First request after client bounding always stuck.

Version 4.28.0

**Comment by zivillian** [comment]
This will coincidentally be fixed by #436

**Comment by zivillian** [comment]
@object If I'm correct this was fixed by #436.

**Comment by object** [comment]
This looks good. I am releasing 4.29 now.
